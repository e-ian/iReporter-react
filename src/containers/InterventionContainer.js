import React, { Component } from "react";
import { connect } from "react-redux";
import { getInterventions } from "../actions/InterventionAction";
import InterventionComponent from "../components/InterventionComponent";
import NavbarComponent from "../components/navBarComponent";

export class interventionContainer extends Component {
  constructor(props) {
    super(props);
  }
  componentDidMount() {
    const { getInterventions } = this.props;
    getInterventions();
  }

  render() {
    const { interventions } = this.props;
    return (
      <div>
        <NavbarComponent />
        <InterventionComponent interventions={interventions} />
      </div>
    );
  }
}
export const mapStateToProps = state => ({
  interventions: state.incident.interventions_list
});
export default connect(
  mapStateToProps,
  { getInterventions }
)(interventionContainer);
