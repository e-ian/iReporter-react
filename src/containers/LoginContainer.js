import React, { Component } from "react";
import { connect } from "react-redux";

import Login from "../components/LoginComponent";
import { userLoginRequest } from "../actions/LoginAction";

export class LoginContainer extends Component {
  constructor(props) {
    super(props);

    this.state = {
      username: "",
      password: "",
      isLoading: false,
      errors: {
        username: "",
        password: ""
      }
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.loginSuccess === true) {
      const { history } = this.props;

      history.push("/interventions");
    }
  }

  handleChange(event) {
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  handleSubmit(event) {
    event.preventDefault();
    this.setState({
      errors: {
        username: "The username is required"
      }
    });
    const data = {
      username: this.state.username,
      password: this.state.password
    };
    this.props.userLoginRequest(data);
  }

  render() {
    return (
      <div>
        <Login
          handleChange={this.handleChange}
          username={this.state.username}
          isProcessing={this.props.isProcessing}
          password={this.state.password}
          errors={this.state.errors}
          handleSubmit={this.handleSubmit}
        />
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  isProcessing: state.auth_login.isProcessing,
  loginSuccess: state.auth_login.loginSuccess
});

export default connect(
  mapStateToProps,
  { userLoginRequest }
)(LoginContainer);
