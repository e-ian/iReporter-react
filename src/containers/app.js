import React, { Component } from "react";
import { Provider } from "react-redux";
import Routes from "../components/routes";
import "bootstrap/dist/css/bootstrap.min.css";
import "react-toastify/dist/ReactToastify.css";
import { ToastContainer } from "react-toastify";

import "../styles/app.scss";
import store from "../store";

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <ToastContainer />
        <Routes />
      </Provider>
    );
  }
}

export default App;
