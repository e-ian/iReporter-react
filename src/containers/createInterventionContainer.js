import React, { Component } from "react";
import { connect } from "react-redux";
import CreateIntervention from "../components/createInterventionComponent";
import { createActions } from "../actions/createInterventionActions";
import NavbarComponent from "../components/navBarComponent";

class CreateInterventionContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      incidentType: "",
      location: "",
      status: "",
      image: "",
      Comment: ""
    };
  }
  componentWillReceiveProps(nextProps) {
    if (nextProps.isRedirecting === true) {
      const { history } = this.props;

      history.push("/interventions");
    }
  }
  onChange = event => {
    const { value, name } = event.target;
    this.setState({ [name]: value });
  };
  OnSubmit = event => {
    event.preventDefault();

    const { incidentType, location, status, image, Comment } = this.state;

    const data = {
      incident_type: incidentType,
      location: location,
      status: status,
      image: image,
      comment: Comment
    };

    this.props.createActions(data);
  };

  render() {
    const { incidentType, location, status, image, Comment } = this.state;
    return (
      <div>
        <NavbarComponent />
        <CreateIntervention
          onChange={this.onChange}
          incidentType={incidentType}
          location={location}
          status={status}
          image={image}
          Comment={Comment}
          OnSubmit={this.OnSubmit}
        />
      </div>
    );
  }
}
export const mapStateToProps = state => ({
  isRedirecting: state.create.isRedirecting
});
// export default CreateInterventionContainer;
export default connect(
  mapStateToProps,
  { createActions }
)(CreateInterventionContainer);
