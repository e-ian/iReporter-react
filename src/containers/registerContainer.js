import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import RegisterComponent from "../components/registerComponent";
import { registerUser } from "../actions/registerActions";
import { passwordValidate, validEmailRegex } from "../utils/regex";
import { validateForm } from "../utils/validateforms";

export class RegisterContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: "",
      firstname: "",
      lastname: "",
      role: "",
      email: "",
      password: "",
      errors: {
        username: "",
        email: "",
        password: ""
      },
      isLoading: false
    };
  }

  componentWillReceiveProps = nextProps => {
    if (nextProps.isLoginSuccess === true) {
      const { history } = this.props;
      history.push("/login");
    }
  };
  handleOnchange = event => {
    // console.log(this.state);

    event.preventDefault();
    let errors = this.state.errors;
    const { name, value } = event.target;
    switch (name) {
      case "username":
        errors.username =
          value.length < 4 ? " username must be atleast 4 characters" : "";
        break;
      case "firstname":
        errors.firstname =
          value.length < 4 ? " firstname must be atleast 4 characters" : "";
        break;
      case "lastname":
        errors.lastname =
          value.length < 4 ? " lastname must be atleast 4 characters" : "";
        break;
      case "role":
        errors.role =
          value.length < 4 ? " role must be either admin or user" : "";
        break;
      case "email":
        errors.email = !validEmailRegex.test(value) ? " Invalid email" : "";
        break;
      case "password":
        errors.password = !passwordValidate.test(value)
          ? "Password must have least a number, lowercase char, special char and min of 8 chars"
          : "";
        break;
      default:
        return null;
    }
    this.setState({ errors, [name]: value });
  };
  handleOnSubmit = event => {
    event.preventDefault();
    if (validateForm(this.state.errors)) {
      const {
        username,
        password,
        email,
        firstname,
        lastname,
        role
      } = this.state;
      const { registerUser, history } = this.props;
      const data = {
        username: username,
        password: password,
        email: email,
        firstname: firstname,
        lastname: lastname,
        role: "admin"
      };

      registerUser(data, history);
    } else {
      return false;
    }
  };

  render() {
    const {
      username,
      password,
      email,
      errors,
      firstname,
      lastname,
      role
    } = this.state;
    return (
      <div>
        <RegisterComponent
          username={username}
          firstname={firstname}
          lastname={lastname}
          role={role}
          isRegistering={this.props.isRegistering}
          password={password}
          email={email}
          errors={errors}
          handleOnChange={this.handleOnchange}
          handleOnSubmit={this.handleOnSubmit}
          error_message={this.props.error}
        />
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  success: state.auth.success,
  error: state.auth.error,
  isRegistering: state.auth.isRegistering,
  isLoginSuccess: state.auth.isLoginSuccess
});
RegisterContainer.propTypes = {
  success: PropTypes.object,
  error: PropTypes.object,
  registerUser: PropTypes.func
};
export default connect(
  mapStateToProps,
  { registerUser }
)(RegisterContainer);
