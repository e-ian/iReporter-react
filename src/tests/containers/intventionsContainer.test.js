import React from "react";
import { interventionContainer } from "../../containers/InterventionContainer";
import { shallow } from "enzyme";

const props = {
  getInterventions: jest.fn()
};
let wrapper;
describe("InterventionContainer", () => {
  beforeEach(() => {
    wrapper = shallow(<interventionContainer {...props} />);
  });
  it("description", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
