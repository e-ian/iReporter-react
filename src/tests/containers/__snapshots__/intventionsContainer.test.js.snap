// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InterventionContainer description 1`] = `ShallowWrapper {}`;
